### DISCUSSION TITLE
in arm after running lldpcli it hangs and not able to connect to lldpd

### DISCUSSION THREAD

**Discussion by mdmuazzamhusain** [lldpd/lldpd#633] [Q&A]
in arm after running lldpcli it hangs and not able to connect to lldpd. So how to fix it. 
In x86 same code is working. just change is platform.

**Comment by mdmuazzamhusain** [comment]
Our lldpcli was getting stuck at malloc allocation, and after usage of calloc , at least we have moved ahead. 
but problem still persist.
log as below:-

# ./sbin/lldpcli -ddd
1970-01-06T23:47:51 [ DBG/lldpctl] connect to lldpd
1970-01-06T23:47:51 [ DBG/lldpctl] conn is not NULL  
1970-01-06T23:47:51 [ DBG/control] send a message through control socket
1970-01-06T23:47:51 [ DBG/control] connect to control socket /run/lldpd.socket
after this cli is stuck.

**Comment by mdmuazzamhusain** [comment]
Hi @vincentbernat 
Is there anything which you can highlight on the above issue.

**Comment by mdmuazzamhusain** [comment]
Hi @vincentbernat  , in x86 without lldpcli execution, lldpd daemon at least sends out default pkts. But in ARM even default pks are not being sent.

**Reply by vincentbernat** [reply]
With so little information, there is not much to do to help. Given the current userbase of lldpd, if it wasn't working on ARM, we would have some reports about that. I don't see how malloc could block. Maybe there is something wrong with your toolchain.

**Reply by mdmuazzamhusain** [reply]
Hi @vincentbernat 
First of all thank you very much for your reply!

Observations on x86 and ARM(FSM Soc From Qualcomm)
1. On x86 just i ran lldpd and started capturing wireshark. I observed lldp packets are being transmitted with the infor what the daemon can gather on it's own since i haven't provided any cli commands either via lldpcli or .conf file.
2. On ARM I expect same, but i don't see any pkt of lldp goes out. This indicates somehow lldpd is not fully functional on arm.
3. Regarding malloc, using same toolchain i have written an malloc example and was working fine. 

Steps to configure and compile for ARM
---------------------------------------------
1. ./configure --prefix=/usr/local/ --host=arm-linux CC=arm-none-linux-gnueabi-gcc --sysconfdir=/etc  --with-embedded-libevent --without-xml  --with-privsep-user=root --with-privsep-group=root 
2. make and make install
3. Now i run simple /usr/local/sbin/lldpd and expect some pkt to go out of interface, but not going. 
4. Toolchain name - Sourcery_CodeBench_Lite_for_ARM_GNU_Linux/bin/arm-none-linux-gnueabi-gcc
5. lldpcli is not able to connect lldpd socket, i see event is not getting generated which actually calls accept syscall.

Please suggest further, would be highly appreciated.

**Reply by vincentbernat** [reply]
You'll need to enable debugging on lldpd side. It is likely the one being stuck.
